src/hooks/sounds.tsx

```diff
@@ -35,33 +35,40 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoaded: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoaded: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
 export const CustomSoundsProvider: React.FC = ({ children }) => {
+  const [isLoaded, setIsLoaded] = useState<boolean>(initialValue.isLoaded);
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
 
   // Fetch the sound configs data on first load
   useEffect(() => {
```

Comment:
While this will only currently run on first render, since it's dependent on a user being logged in, it might make sense to add that to it's dependencies.

We could probably use `user` from `src/hooks/useUser.ts` for this.

---

Alternatively, we could follow the default pattern described in the firebase docs, though it sort of deviates from our existing patterns a little (though isn't a bad pattern necessarily):

- https://firebase.google.com/docs/auth/web/manage-users#get_the_currently_signed-in_user

To do that, we would probably have something like this:

```typescript
const [isLoggedIn, setLoggedIn] = useState(false);

useEffect(() => {
  const unsubscribeListener = firebase.auth().onAuthStateChanged((user) => {
    if (user) {
      // User is signed in.
      setLoggedIn(true);
    } else {
      // No user is signed in.
      setLoggedIn(false);
    }
  });

  return () => {
    unsubscribeListener()
  }
}, [])
```

We could then use `isLoggedIn` as the dependency that controls whether we try to fetch the data or not. This should make it work even if this provided is positioned 'outside' of the `AuthIsLoaded` component.

If we do this, it would probably make sense to wrap it up as it's own custom hook, then call it here.

Reply:
I prefer using `user` from `src/hooks/useUser.ts`